core/src/test/scala/unit/kafka/tools/DumpLogSegmentsTest.scala

```diff
@@ -243,6 +244,32 @@ class DumpLogSegmentsTest {
     assertEquals(Map.empty, errors.shallowOffsetNotFound)
   }
 
+  @Test
+  def testDumpRemoteLogMetadataRecords(): Unit = {
```

Comment:
please add the following test cases:

1. metadata is empty
2. metadata contains one record, one batch
3. metadata contains multiple records (testing with 2 is fine), one batch
4. metadata contains multiple records, multiple batches
5. starting record of metadata log is at non-zero offset
6. metadata is corrupted (maybe because let's say batch header doesn't match number of records)
7. IOException occurs when reading from segment
8. the cli contains --remote-log-metadata-decoder without specifying the second flag "--files"
9. there is no file present at path provided by --files logFilePath (different from case 7 where file exists but we get IOException while reading)

Additionally, 

10. do we compact metadata? If yes, can you add cases where segments is a compacted segment (has some offsets missing).
11. do we compress metadata? If test, can you add cases which validate correct deserialization for different compression types
12. please assert that the correct full record (for at least one of them) is printed instead of just checking whether it contains COPY_SEGMENT_FINISHED

Reply:
Hi @divijvaidya, thanks. Most of the suggested tests are implemented now.

> 3. metadata contains multiple records (testing with 2 is fine), one batch

I guess this is the original test.

> do we compact metadata? If yes, can you add cases where segments is a compacted segment (has some offsets missing).

No, the cleanup policy for this topic is hard coded to delete. See [here](https://github.com/apache/kafka/blob/3.7.1/storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManager.java#L491).

> do we compress metadata? If test, can you add cases which validate correct deserialization for different compression types

No, they are not compressed. See for example [here.](https://github.com/apache/kafka/blob/3.7.1/core/src/main/java/kafka/log/remote/RemoteLogManager.java#L746-L750)